### ISSUE TITLE
Sony PlayStation 4 Dualshock mapping

### ISSUE THREAD

**Issue by tristan-k** [mupen64plus/mupen64plus-ui-python#67]
Does someone have a proper mapping for the PS4 Dualshock? I started editing `InputAutoCfg.ini`in `/Applications/M64Py.app/Contents/Resources/InputAutoCfg.ini` on Mac OS X with `M64Py 0.2.3` but the changes are not applied. 

```
$ lsusb
Bus 029 Device 007: ID 054c:05c4 Sony Corporation Wireless Controller 
```

```
[PLAYSTATION(R)4 Controller]
[Sony PLAYSTATION(R)4 Controller]
[Sony Computer Entertainment Wireless Controller]
[Sony Corporation Wireless Controller]
[Wireless Controller]
plugged = True
plugin = 2
mouse = False
AnalogDeadzone = 4096,4096
AnalogPeak = 32768,32768
DPad R = button(5)
DPad L = button(7)
DPad D = button(6)
DPad U = button(4)
Start = button(3)
Z Trig = button(10)
B Button = button(15)
A Button = button(14)
C Button R = axis(2+)
C Button L = axis(2-)
C Button D = axis(3+)
C Button U = axis(3-)
R Trig = button(11)
L Trig = button(8)
Mempak switch =
Rumblepak switch =
X Axis = axis(0-,0+)
Y Axis = axis(1-,1+)
```

**Comment by gen2brain** [comment]
Try with new release.

**Comment by martinezjorge** [comment]
So I'm trying to do this exact same thing but every combination of configurations has been unsuccessful. One thing that I noticed is that when I try to have all these Sections the next time that I open the cfg file all the Sections are separated after everything gets alphabetized. 

@gen2brain from your response it seems that you expected this configuration to work. I outlined some of the stuff that I did in this reddit post.

https://www.reddit.com/r/linux_gaming/comments/i9gl3n/trouble_connecting_ds4_to_mupen64_on_ubuntu_1804/

As the post states, I'm using Ubuntu 18.04 and a DS4 so I'm not really sure where I'm going wrong. The OS recognizes the control and when I use jscal it can see the joysticks moving and buttons being pressed so I think once I can get the controller connected to mupen the key mappings shouldn't be too complicated.

Any help appreciated.
